x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.ts

```diff
@@ -360,7 +360,7 @@ export function registerIndexRoutes({
           pipelineName,
           modelId,
           sourceField,
-          destinationField || modelId,
```

Comment:
It's not the API tier's responsibility to make this decision. Moving it down to the app tier.